ostd/src/arch/x86/boot/smp.rs

```diff
@@ -186,17 +191,27 @@ unsafe fn fill_boot_info_ptr(info_ptr: *const PerApRawInfo) {
 
 /// # Safety
 ///
-/// The caller must ensure the pointer to be filled is valid to write.
```

Comment:
I still think this is accurate. You need to prove this (e.g., the APs are not booted so there is no data race). The proof should be done in the initialization method, not in `fill_boot_pt_ptr`.